File: 0226_invert_binary_tree/iterative_bfs/kotlin/solution.kt
class TreeNode(
    var `val`: Int,
    var left: TreeNode? = null,
    var right: TreeNode? = null,
)

fun invertTree(root: TreeNode?): TreeNode? {
    if (root == null) return null

    val queue = ArrayDeque<TreeNode>()
    queue.addLast(root)

    while (queue.isNotEmpty()) {
        val node = queue.removeFirst()
        val left = node.left
        node.left = node.right
        node.right = left

        node.left?.let { queue.addLast(it) }
        node.right?.let { queue.addLast(it) }
    }

    return root
}

File: 0226_invert_binary_tree/iterative_bfs/python/solution.py
from __future__ import annotations

from collections import deque
from dataclasses import dataclass


@dataclass
class TreeNode:
    val: int
    left: TreeNode | None = None
    right: TreeNode | None = None


def invert_tree(root: TreeNode | None) -> TreeNode | None:
    if root is None:
        return None

    queue: deque[TreeNode] = deque([root])
    while queue:
        node = queue.popleft()
        node.left, node.right = node.right, node.left

        if node.left is not None:
            queue.append(node.left)
        if node.right is not None:
            queue.append(node.right)

    return root
